modules/spring-config/src/main/java/org/mule/runtime/config/internal/SpringXmlConfigurationBuilder.java

```diff
@@ -256,19 +264,26 @@ private ArtifactAst createApplicationModel(Set<ExtensionModel> extensions,
   }
 
   private AstXmlParser createMuleXmlParser(Set<ExtensionModel> extensions,
-                                           Map<String, String> artifactProperties, boolean disableXmlValidations) {
+                                           Map<String, String> artifactProperties, boolean disableXmlValidations,
+                                           MuleContext muleContext)
+      throws MuleException {
     ConfigurationPropertiesResolver propertyResolver =
         new DefaultConfigurationPropertiesResolver(empty(), new StaticConfigurationPropertiesProvider(artifactProperties));
 
+    muleContext.getInjector().inject(this);
```

Comment:
Cant't we do this injection at an earlier point (I'm thinking in the doConfigure method) and avoid passing the muleContext around?